### ISSUE TITLE
Feature request: Move the control for the number of columns to the block toolbar

### ISSUE THREAD

**Issue by carolinan** [WordPress/gutenberg#30991]
## What problem does this address?

This was brought up during a user testing session. People expected to be able to change the number of columns in the toolbar and not in the sidebar.

## What is your proposed solution?

Move the setting to the toolbar.

**Comment by carolinan** [comment]
@WordPress/gutenberg-design Hi, any thoughts about this request? I do not have enough contact with users to know if this is a frequent problem.

**Comment by jasmussen** [comment]
Just to be sure, you're referring to adding a button to the Columns block toolbar?

<img width="1392" alt="columns block with block toolbar" src="https://github.com/WordPress/gutenberg/assets/1204802/86442156-3386-4d4d-8b03-52dcd783c761">

If so, sure I could see it happening, but it opens a number of questions:

* Is it a "plus" icon button, and if so, do we then also add a minus button to remove columns?
* Is it a stepped input, and what's the accessibility story there? 
* Is there a better way to add/remove columns, perhaps in the canvas directly? What are other common patterns for creating columns?
* Is there a closer relationship with the group block, so perhaps even the group button has the plus to add additional columns? Or is it valid to have a single-column columns block?

For me the biggest issue with the columns block is that it's layered/nested, and in that light I'm not sure adding controls to the block toolbar would help. Because you'd still have to select the right nested block in order for it to work. Given Columns is also one of the older blocks we have, I wonder if there's a way to modernize it, both in UI and utility? 

I know that @jameskoster and @SaxonF  have both thought a lot about simplifying our container blocks, and how layout works across. I wonder if there are some learnings from those efforts we can apply here.

**Comment by jameskoster** [comment]
For me it depends on how we see Columns evolving in the context of https://github.com/WordPress/gutenberg/issues/42385. 

If we think about the Columns experience transitioning to a Row of Stacks then the inherent method for inserting 'columns' seems a bit more intuitive compared with a toolbar or inspector control:

<img width="1145" alt="Screenshot 2023-07-24 at 15 37 47" src="https://github.com/WordPress/gutenberg/assets/846565/1a69105a-adfb-48d5-a6cd-9f69067c73fa">

I've always found the input/range a bit awkward for _removing_ columns due to how it wipes out your content.

@SaxonF will certainly have deeper insights as he's thought about layout a lot :)

**Comment by carolinan** [comment]
See https://github.com/WordPress/gutenberg/issues/30716

**Comment by Mamaduka** [comment]
I just remembered a recent and related discussion - #67843. Maybe we should combine those into a single issue?

**Comment by carolinan** [comment]
Yes lets close this issue since it has the least activity and visibility. I am just going through the 6 pages of issues I have opened...

**Comment by Mamaduka** [comment]
> I am just going through the 6 pages of issues I have opened...

Stay strong, @carolinan!
